packages/expo-file-system/android/src/main/java/expo/modules/filesystem/unifiedfile/AssetFile.kt

```diff
@@ -0,0 +1,120 @@
+package expo.modules.filesystem.unifiedfile
+
+import android.content.Context
+import android.net.Uri
+import android.webkit.MimeTypeMap
+import androidx.core.net.toUri
+import java.io.File
+import java.io.InputStream
+import java.io.OutputStream
+
+class AssetFile(private val context: Context, override val uri: Uri) : UnifiedFileInterface {
+  val path: String = uri.path?.trimStart('/') ?: throw IllegalArgumentException("Invalid asset URI: $uri")
+
+  override fun exists(): Boolean = isDirectory() || isFile()
+
+  override fun isDirectory(): Boolean {
+    return context.assets.list(path)?.isNotEmpty() == true
+  }
+
+  override fun isFile(): Boolean {
+    return try {
+      context.assets.open(path).use {
+        true
+      }
+    } catch (_: Exception) {
+      false
+    }
+  }
+
+  override val parentFile: UnifiedFileInterface?
+    get() {
+      val currentPath = uri.path.orEmpty()
+      if (currentPath.isEmpty()) {
+        return null
+      }
```

Comment:
```suggestion
      val currentPath = uri.path ?: return null
```
I don't see the value in checking if the path is empty. Passing an empty URL should be detected during the construction of the file as an incorrect argument

Reply:
I mean, if you call the parentFile on a root directory it should return null, we shouldn't event call the constructor with invalid arguments (?)